.docker/images/nginx/no_robots_govcms_host.conf

```diff
@@ -0,0 +1,13 @@
+# Return with a `Disallow: /` for non-production URLs
+
+if ($uri = "/robots.txt"){
+  set $robots_disallow_test "isRobotstxtURI";
+}
+# matches *.govcms.gov.au (also multiple subdomains) except www.govcms.gov.au
+if ($host ~* ^(?!www\.govcms\.gov\.au$)(?:[\w\-]+\.)+govcms\.gov\.au$){
+  set $robots_disallow_test "$robots_disallow_test+isGovcmsHost";
+}
+
```

Comment:
This just feels like a lot of logic to do a simple thing.

Reply:
those rules are from https://github.com/govCMS/govcmslagoon/blob/master/.docker/images/nginx/no_robots_govcms_host.conf

Reply:
How does this functionality interact with
https://github.com/govCMS/govcms8lagoon/blob/master/.docker/images/nginx/no-robots.sh

Why do we have a neat set of helpers here
https://github.com/govCMS/govcms8lagoon/tree/master/.docker/images/nginx/helpers
yes this is called `no_robots_govcms_host.conf` in a different directory.

Reply:
`no-robots.sh` is for non-prod env.
`no_robots_govcms_host.conf` is for PROD when accessed from the editorial domain `*.govcms.gov.au`.

Reply:
Since no-robots is pretty clear in its naming and seems perfectly named for this new functionality, and since it already exhibits an example of conditionally applying "disallow in robots", and has the benefit of using bash logic and not nginx whacky logic, and since it already exists... can you please extend no-robots.sh with an extra logic check where the site is on an editorial domain *.govcms.gov.au.

Reply:
`no-robots.sh` does not run in the context of Nginx. It runs during the Lagoon deployment and outputs the robots.txt for non-production Lagoon environments.

On the other hand, `no_robots_govcms_host.conf` is an Nginx config to output the alternative robots.txt if the hostname is `*.govcms.gov.au` on Production environment. The purpose of this config is to prevent search engines to index the `*.govcms.gov.au` domain of the Production sites.

Reply:
> It runs during the Lagoon deployment.
The new functionality also runs doing lagoon deployments.

>outputs the robots.txt for non-production Lagoon environments.
The new functionality also applies to non-production environments, it just changes the scope a little.

I feel certain that this change can be done with a modification to existing no-robots.sh.

Reply:
Lagoon deployment has no context of on which URL Nginx is being served. Once the Production env is successfully built, it can be accessed via 2 different URLs: `project.gov.au` and `project.govcms.gov.au`. When being accessed via `project.gov.au`, Nginx serves the site's robots.txt, and on `project.govcms.gov.au` it should deny all search engines.

When Nginx is running there is no Lagoon deployment.

Reply:
I think it will look like this:

```
#!/bin/sh
##
# Add robots.txt on all non production environments, including prod sites on *.govcms.gov.au domains.
#

if [ ! "${LAGOON_ENVIRONMENT_TYPE}" == "production" ]; then
    printf "User-agent: *\nDisallow: /\n" > /app/web/robots.txt
fi

if [ "${LAGOON_WHATEVER_THE_DOMAIN_IS}" = "big old regex" ]; then
    printf "User-agent: *\nDisallow: /\n" > /app/web/robots.txt
fi

```

Reply:
One Lagoon env can have multiple routes. Only one route should have the normal robots.txt, the others should deny all bots.

Reply:
Aha nice but also this is not enough then. If we can have multiple routes then what happens if there is an additional route that doesn't match *govcms.gov.au? This doesn't protect from this.

I think this is ok then, but can you add a comment in the .conf ... Something like:

"A lagoon production environment can have multiple routes, disallow robots.txt for any *.govcms.gov.au domains. This is a fallback mechanism, since a site should globally redirect non-production routes to their canonical production route. This will not pick up duplicate content issues for additional routes that don't match *.govcms.gov.au ."

Reply:
Default Prod Lagoon env has 2 routes: the live domain and the editorial govcms domain. The purpose of this PR is to ensure that search engines do not index content of the editorial domain. If a GovCMS customer wants extra domains added to the Production env, it's their business decision on how to handle duplicate content across their domains in terms of SEO.

Reply:
Have you also considered using these variables:

LAGOON_ROUTE
LAGOON_ROUTES

LAGOON_ROUTE is the main route, so is it possible to just add robots.txt if the current route doesn't match the main route?

Reply:
That aside, if you put a bunch of this information you're supplying in the .conf I think it's good to go. There is just a lot of business logic that applies here that needs to be captured.

Reply:
Please note that those config are already on GovCMS7 on Lagoon. This PR is just to align the two versions of GovCMS.

Any extra modifications need to be considered in the context of both versions.

Reply:
I understand. It's just really hard to change things after they go in. And we need better commenting in general. If you could PR the same comments in the d7 then that would be good.

Reply:
I need to follow up on some stuff relating to this. I will return :)

Reply:
@sonnykt sorry for the wait. Can you please put the comment "Prevent robot crawling of this domain when the *.govcms.gov.au editorial domain is configured to be accessible in addition to the public domain." It should really be added against d7 as well.